package/base-files/files/lib/upgrade/nand.sh

```diff
@@ -270,7 +270,12 @@ nand_upgrade_ubinized() {
 	nand_detach_ubi "$CI_UBIPART" || return 1
 
 	local mtdnum="$( find_mtd_index "$CI_UBIPART" )"
-	${gz}cat "$ubi_file" | ubiformat "/dev/mtd$mtdnum" -y -f - && ubiattach -m "$mtdnum"
+	if [ -n "$gz" ]; then
+		${gz}cat "$ubi_file" | ubiformat "/dev/mtd$mtdnum" -y -f - && ubiattach -m "$mtdnum"
+	else
+		# On some bcm53xx devices, the cat command leads to unintended results
+		ubiformat "/dev/mtd$mtdnum" -y -f "$ubi_file" && ubiattach -m "$mtdnum"
```

Comment:
On which devices do you see this problem?
When `nand_upgrade_ubinized` is called with the 2. parameter set to an empty string it should do the same as before.

Reply:
On PHICOMM K3, use `cat "$ubi_file" | ubiformat ... `does not work, 
when I write as `ubiformat "/dev/mtd$mtdnum" -y -f "$ubi_file"` does